backend/src/controllers/reports/helpers/index.js

```diff
@@ -152,19 +152,34 @@ const getHeaderInfoFrom = (templateType = "docx") => {
 const getReport = getReportAndSetRequestHeaders();
 
 /**
- * Groups an array of objects by a specified property.
+ * Groups items in a list by a specified property.
  *
- * @param   {object[]} rows - The array of objects to group.
- * @param   {string}   prop - The property to group by.
- * @returns {object[]}      Returns an array of objects grouped by the specified property.
+ * @param   {Array}  list  - The list of items to be grouped.
+ * @param   {string} group - The property name to group the items by. Default is "portfolio_name".
+ * @returns {Array}        An array of objects, where each object represents a group and its associated items.
+ *                         Each object has two properties: the specified 'group' property (key) and the array of items (value).
  */
-const groupByProperty = (rows, prop) =>
-  _.isEmpty(rows)
-    ? rows
-    : _.map(_.groupBy(rows, prop), (value, key) => ({
-        [prop]: key,
-        projects: [...value],
-      }));
+const groupByProperty = (list = [], group = "portfolio_name") => {
+  // Using lodash's chain function "_" to create a chainable sequence.
+  // The input 'list' is wrapped in the chain and will be operated upon.
+  const result = _(list)
+    // Group the items in the list by the specified 'group' property.
+    .groupBy(group)
+    // After grouping, 'map' is used to convert the grouped items into a new format.
+    .map((projects, key) => ({
+      // The 'key' represents the value of the property used for grouping (e.g., portfolio_name).
+      // Create an object with the 'group' property as the key (e.g., { portfolio_name: 'some_group_value' }).
```

Comment:
Nice commenting here!

Reply:
it's way too compact and obscure otherwise, but I plan to cut back on the verbosity eventually ;)